data/collector_types.go

```diff
@@ -0,0 +1,394 @@
+	case CaptureTypeTabular:
+		return datasyncPB.DataType_DATA_TYPE_TABULAR_SENSOR
+	case CaptureTypeBinary:
+		return datasyncPB.DataType_DATA_TYPE_BINARY_SENSOR
+	case CaptureTypeUnspecified:
+		return datasyncPB.DataType_DATA_TYPE_UNSPECIFIED
+	default:
+		return datasyncPB.DataType_DATA_TYPE_UNSPECIFIED
+	}
+}
+
+// GetDataType returns the DataType of the method.
+func GetDataType(methodName string) CaptureType {
+	switch methodName {
+	case nextPointCloud, readImage, pointCloudMap, GetImages, captureAllFromCamera:
+		return CaptureTypeBinary
+	default:
+		return CaptureTypeTabular
+	}
+}
+
+// Timestamps are the timestamps the data was captured.
+type Timestamps struct {
+	// TimeRequested represents the time the request for the data was started
+	TimeRequested time.Time
+	// TimeReceived represents the time the response for the request for the data
+	// was received
+	TimeReceived time.Time
+}
+
+// MimeType represents the mime type of the sensor data.
+type MimeType int
+
+// This follows the mime types supported in
+// https://github.com/viamrobotics/api/pull/571/files#diff-b77927298d8d5d5228beeea47bd0860d9b322b4f3ef45e129bc238ec17704826R75
+const (
+	// MimeTypeUnspecified means that the mime type was not specified.
+	MimeTypeUnspecified MimeType = iota
+	// MimeTypeImageJpeg means that the mime type is jpeg.
+	MimeTypeImageJpeg
+	// MimeTypeImagePng means that the mime type is png.
+	MimeTypeImagePng
+	// MimeTypeApplicationPcd means that the mime type is pcd.
+	MimeTypeApplicationPcd
+)
+
+// ToProto converts MimeType to datasyncPB.
+func (mt MimeType) ToProto() datasyncPB.MimeType {
+	switch mt {
+	case MimeTypeUnspecified:
+		return datasyncPB.MimeType_MIME_TYPE_UNSPECIFIED
+	case MimeTypeImageJpeg:
+		return datasyncPB.MimeType_MIME_TYPE_IMAGE_JPEG
+	case MimeTypeImagePng:
+		return datasyncPB.MimeType_MIME_TYPE_IMAGE_PNG
+	case MimeTypeApplicationPcd:
+		return datasyncPB.MimeType_MIME_TYPE_APPLICATION_PCD
+	default:
+		return datasyncPB.MimeType_MIME_TYPE_UNSPECIFIED
+	}
+}
+
+// MimeTypeFromProto converts a datasyncPB.MimeType to a data.MimeType.
+func MimeTypeFromProto(mt datasyncPB.MimeType) MimeType {
+	switch mt {
+	case datasyncPB.MimeType_MIME_TYPE_UNSPECIFIED:
+		return MimeTypeUnspecified
+	case datasyncPB.MimeType_MIME_TYPE_IMAGE_JPEG:
+		return MimeTypeImageJpeg
+	case datasyncPB.MimeType_MIME_TYPE_IMAGE_PNG:
+		return MimeTypeImagePng
+	case datasyncPB.MimeType_MIME_TYPE_APPLICATION_PCD:
+		return MimeTypeApplicationPcd
+	default:
+		return MimeTypeUnspecified
+	}
+}
+
+// CameraFormatToMimeType converts a camera camerapb.Format into a MimeType.
+func CameraFormatToMimeType(f camerapb.Format) MimeType {
+	switch f {
+	case camerapb.Format_FORMAT_UNSPECIFIED:
+		return MimeTypeUnspecified
+	case camerapb.Format_FORMAT_JPEG:
+		return MimeTypeImageJpeg
+	case camerapb.Format_FORMAT_PNG:
+		return MimeTypeImagePng
+	case camerapb.Format_FORMAT_RAW_RGBA:
+		fallthrough
+	case camerapb.Format_FORMAT_RAW_DEPTH:
+		fallthrough
```

Comment:
`TODO` comments for tracking these?  or if desired, explain why

Reply:
how is that?